### PR TITLE
Feature/resources exception logging

### PR DESCRIPTION
**Pull request by abhishekjain88**
Added logging for exceptions in all REST API resources. No regressions in unit and integration tests.

### CODE REVIEW

**Comment by abhishekjain88** [comment]
@bargenilesh @AndreaSuckro Thank you for informing me about the formatting change. I was using the other one, wasabi_eclipse and that could have caused the mismatch in format. Fixed it now. Requesting review.

**Comment by abhishekjain88** [comment]
@iizrailevsky @jcwuzoegiver A gentle reminder on the review.

File: modules/api/src/main/java/com/intuit/wasabi/api/AuditLogResource.java [hunk 1]
```diff
@@ -116,53 +120,62 @@
...
+            final String sort,
+
+            @QueryParam("timezone")
+            @DefaultValue(DEFAULT_TIMEZONE)
+            @ApiParam(name = "timezone", defaultValue = DEFAULT_TIMEZONE, value = DOC_TIMEZONE)
+            final String timezoneOffset) {
+        try {
+            List<AuditLogEntry> auditLogs;
+
+            if (applicationName != null) {
```
Comment (mans4singh) [inline modules/api/src/main/java/com/intuit/wasabi/api/AuditLogResource.java]: Some debug logging will help.

Comment (abhishekjain88) [inline modules/api/src/main/java/com/intuit/wasabi/api/AuditLogResource.java]: 👍 on that @mans4singh ! We should do it as part of a separate task. This task mostly caters to error scenarios.

**Comment by mans4singh** [comment]
@abhishekjain88 - I think some more debug logging will help in the long run.

**Comment by abhishekjain88** [comment]
@mans4singh I agree. This should be added to all APIs in general. I suggest we do this incrementally by pushing this change in and adding debug logging as a separate change.

File: modules/analytics-objects/src/main/java/com/intuit/wasabi/analyticsobjects/EventList.java [hunk 2]
```diff
@@ -38,4 +38,12 @@ public EventList() {
     public void setEvents(List<Event> value) {
         events = value;
     }
+
+    @Override
+    public String toString() {
+        if (null != events) {
```
Comment (jcwuzoegiver) [inline modules/analytics-objects/src/main/java/com/intuit/wasabi/analyticsobjects/EventList.java]: can possibly do if (!isNull(events))

File: modules/authorization-objects/src/main/java/com/intuit/wasabi/authorizationobjects/UserRoleList.java [hunk 3]
```diff
@@ -72,4 +72,12 @@ public int hashCode() {
                 .append(getRoleList().size())
                 .toHashCode();
     }
+
+    @Override
+    public String toString() {
+        if (null != roleList)
```
Comment (jcwuzoegiver) [inline modules/authorization-objects/src/main/java/com/intuit/wasabi/authorizationobjects/UserRoleList.java]: use !isNull()?

File: modules/analytics-objects/src/main/java/com/intuit/wasabi/analyticsobjects/EventList.java (hunk 2 above)
Comment (abhishekjain88) [inline modules/analytics-objects/src/main/java/com/intuit/wasabi/analyticsobjects/EventList.java]: 👍 Addressed!

File: modules/authorization-objects/src/main/java/com/intuit/wasabi/authorizationobjects/UserRoleList.java (hunk 3 above)
Comment (abhishekjain88) [inline modules/authorization-objects/src/main/java/com/intuit/wasabi/authorizationobjects/UserRoleList.java]: 👍 Addressed!

**Comment by bargenilesh** [comment]
@abhishekjain88 Changes looks good, provided functional tests are passing :shipit:

**Comment by jcwuzoegiver** [comment]
@abhishekjain88 :shipit: provided it passes all functional tests.

**Comment by iizrailevsky** [comment]
👍 :shipit:

**Comment by mans4singh** [comment]
Looks good :shipit:

**Pull request merged**